app/views/administration/viewrole.html

```diff
@@ -47,17 +47,46 @@ <h3 class="bolder">{{'label.heading.enablerole' | translate}}</h3>
             <button class="btn btn-primary" ng-click="enableRoles()">{{'label.button.confirm' | translate}}</button>
         </div>
     </script>
+    <script type="text/ng-template" id="editName.html">
+		<div class="modal-header silver">
+			<h3 class="bolder">{{'label.edit' | translate}}</h3>
```

Comment:
Add an `<api-validate></api-validate>` here, since there might be errors when you are editing name. For eg: Try to edit the name to an existing role and permission, the platform will return an error in that case. That error needs to be displayed in the modal.